src/test/rgw/test_d4n_filter.cc

```diff
@@ -0,0 +1,2810 @@
+
+    conn->async_exec(req, resp, yield[ec]);
+
+    ASSERT_EQ((bool)ec, false);
+    EXPECT_EQ((int)std::get<0>(resp).value(), 1);
+    EXPECT_EQ((int)std::get<1>(resp).value(), 1);
+    EXPECT_EQ((int)std::get<2>(resp).value(), 1);
+    EXPECT_EQ((int)std::get<3>(resp).value(), 1);
+    EXPECT_EQ((int)std::get<4>(resp).value(), 1);
+    EXPECT_EQ((int)std::get<5>(resp).value(), 1);
+    EXPECT_EQ(std::get<6>(resp).value()[0], instance);
+    EXPECT_EQ(std::get<6>(resp).value()[1], "null");
+    EXPECT_EQ(std::get<7>(resp).value().size(), 10);
+    EXPECT_EQ(std::get<8>(resp).value().size(), 10);
+    EXPECT_EQ(std::get<9>(resp).value().size(), 10);
+    EXPECT_EQ(std::get<10>(resp).value().size(), 10);
+    EXPECT_EQ(std::get<11>(resp).value().size(), 10);
+
+    std::string version = std::get<12>(resp).value();
+    std::error_code err;
+    std::string testData; 
+    std::ifstream testFile; 
+
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/" + instance, err), true);  
+    std::string oid = instance + "#0#" + std::to_string(ofs);
+    std::string location = CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/" + oid;
+    EXPECT_EQ(fs::exists(location, err), true);     
+
+    testFile.open(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/" + oid);
+    ASSERT_EQ(testFile.is_open(), true);
+    getline(testFile, testData);
+    EXPECT_EQ(testData, "test data");
+    testFile.close();
+
+    EXPECT_EQ(fs::exists(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/" + version, err), true);  
+    oid = version + "#0#" + std::to_string(ofs);
+    location = CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/" + oid;
+    EXPECT_EQ(fs::exists(location, err), true);     
+    
+    testFile.open(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + TEST_OBJ + testName + "/" + oid);
+    ASSERT_EQ(testFile.is_open(), true);
+    getline(testFile, testData);
+    EXPECT_EQ(testData, "test data");
+    testFile.close();
+
+    testFile.close();
+    conn->cancel();
+    testBucket->remove(env->dpp, true, optional_yield{yield});
+    DriverDestructor driver_destructor(static_cast<rgw::sal::D4NFilterDriver*>(driver));
+  }, rethrow);
+
+  io.run();
+}
+
+TEST_F(D4NFilterFixture, CopyNoneVersionedObjectWrite) // TODO: check version value if it's expected
+{
+  env->cct->_conf->d4n_writecache_enabled = true;
+  const std::string testName = "CopyNoneVersionedObjectWrite";
+ 
+  boost::asio::spawn(io, [this, &testName] (boost::asio::yield_context yield) {
+    init_driver(yield);
+    create_bucket(testName, yield);
+    std::string instance; 
+    put_version_enabled_object(testName, instance, yield);
+
+    RGWEnv rgw_env;
+    req_info info(env->cct.get(), &rgw_env);
+    rgw_zone_id zone;
+    rgw_placement_rule placement;
+    ceph::real_time mtime;
+    buffer::list bl;
+    bl.append("dest_object_version");
```

Comment:
again here dest_version will be calculated  in the driver and is not taken from the ID tag as explained in one of the previous tests.